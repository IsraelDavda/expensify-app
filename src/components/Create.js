import React from 'react';

const Create = () => (
    <div>This is from Create page</div>
);
export default Create;