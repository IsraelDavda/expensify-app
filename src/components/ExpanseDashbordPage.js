import React from 'react';
import ExpenseList from './expenseList';
import ExpenseFiltersList from './expenseFiltersList';
import ExpensesSummary from './ExpenseSummary';


const ExpanseDashbordPage = () => (
    <div>
    <ExpensesSummary />
    <ExpenseFiltersList />
    <ExpenseList />
    </div>
);

export default ExpanseDashbordPage;