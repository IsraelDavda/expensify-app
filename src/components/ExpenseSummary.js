// rendering by expense dashbaord
// connecting to the store for:
// -expense count (how many expense )visable
// - expense total (how many total is)
import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expensess';
import selectExpensesTotal from '../selectors/expensesTotal';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal /100).format('$0,0.00');

    return (
        <div>
            <h1> Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return{
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};
export default connect(mapStateToProps)(ExpensesSummary)
