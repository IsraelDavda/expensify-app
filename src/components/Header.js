import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/Auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1> Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact>Dashboard </NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense </NavLink>
        <button onClick={startLogout}>Logout</button>
        {/* <NavLink to="/expense" activeClassName="is-active">expense </NavLink> */}
        {/* <NavLink to="/edit" activeClassName="is-active">Go To Edit </NavLink> */}
        {/* <NavLink to="/help" activeClassName="is-active">Go To Help </NavLink> */}
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);