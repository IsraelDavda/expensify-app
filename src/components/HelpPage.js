import React from 'react';

const HelpPage = () => (
    <div>This is from helppage page</div>
);
export default HelpPage;