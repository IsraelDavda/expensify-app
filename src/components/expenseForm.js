import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';



export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            description:props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note: '',
            amount: props.expense ? (props.expense.amount /100).toString(): '',
            createAt:props.expense ? moment(props.expense.createAt) :moment(),
            calenderFocused:false,
            error: ''
        }
    }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState({description})
}
onAreaChange = (e) => {
    const note= e.target.value;
    this.setState({note:note})
}
anountChange = (e) => {
    const amount = e.target.value;
    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
        this.setState(()=>({amount}));
    }
}
onDateChange = (createAt) => {
    if(createAt){
        this.setState(()=>({createAt}))
    }
}

onFocuseChange = ({focused}) => {
    this.setState(()=>({calenderFocused:focused}))
}
onSubmit = (e) => {
    e.preventDefault();
// Set error state eQual to Please provide description and amount
if(!this.state.description || !this.state.amount){
    this.setState(()=> ({error:'Please provide description and amount'}))
}else {
    //Clear the error
    this.setState(()=>({error:''}));
    this.props.onSubmit({
        description:this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createAt: this.state.createAt.valueOf(),
        note: this.state.note
        })
    }
}

    render(){
        return(
            <div>
            {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                    type='text'
                    placeholder='Description'
                    autoFocus
                    value = {this.state.description}
                    onChange={this.onDescriptionChange}
                    />
                    <input
                        type="text"
                        placeholder='Amount'
                        value={this.state.amount}
                        onChange={this.anountChange}
                    />
                    <SingleDatePicker
                    date={this.state.createAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calenderFocused}
                    onFocusChange={this.onFocuseChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    />
                    <textarea
                    placeholder='Add a note to a expnese liat (optional)'
                    onChange={this.onAreaChange}></textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}