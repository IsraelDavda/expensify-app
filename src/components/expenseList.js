import React from 'react';
import { connect } from 'react-redux';
import ShowAll from './expenseListItem';
import Selectexp from '../selectors/expensess';

export const ExpenseList = (props) => (
    <div>
    {
        props.expense.length === 0 ? (
            <p>no expense</p>
        ):(
            props.expense.map((expense)=>{
                return <ShowAll key={ expense.id} {...expense} />

            })
        )
    }
        </div>

        )

const mapStateToProps = (state) =>{
     return{
        expense: Selectexp(state.expenses, state.filters)

    }}



export default connect(mapStateToProps)(ExpenseList);