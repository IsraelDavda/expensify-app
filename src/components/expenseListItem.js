import React from 'react';
import { Link } from 'react-router-dom';


const showAll = ({ id, description, amount, createdAt}) => (
            <div>
            <Link to={`/edit/${id}`}>
                <h3> {description}</h3>
            </Link>
        <p>{amount} - {createAt}</p>
</div>
);

export default showAll;

