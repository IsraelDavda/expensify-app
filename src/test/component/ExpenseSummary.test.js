import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpenseSummary';

test('should currectly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});
test('should currectly render ExpensesSummary with multiply expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={234567898765} />);
    expect(wrapper).toMatchSnapshot();
});