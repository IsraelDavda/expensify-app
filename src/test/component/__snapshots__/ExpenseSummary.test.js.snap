// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should currectly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should currectly render ExpensesSummary with multiply expense 1`] = `
<div>
  <h1>
     Viewing 
    23
     
    expenses
     totalling 
    $2,345,678,987.65
  </h1>
</div>
`;
